extensions/amp-ad/0.1/amp-ad-3p-impl.js

```diff
@@ -144,6 +169,36 @@ export class AmpAd3PImpl extends AMP.BaseElement {
     this.uiHandler = new AmpAdUIHandler(this);
 
     setupA2AListener(this.win);
+
+    this.computeFullWidthState_();
+
+    if (this.fullWidthState_ == FullWidthState.REQUESTED) {
+      return this.attemptFullWidthSizeChange_();
+    }
+  }
+
+  /**
+   * Sets the fullWidthState_ variable.
+   * @private
+   */
+  computeFullWidthState_() {
+    const hasFullWidth = this.element.hasAttribute('data-full-width');
+    const hasFullVW = this.element.getAttribute('width') == '100vw';
```

Comment:
throw use error? If width is not `100vw`

Reply:
Done.